### DISCUSSION TITLE
Prevent another object to move when drawing

### DISCUSSION THREAD

**Discussion by Saqib92** [fabricjs/fabric.js#8720] [Q&A] [answered]
I have an image already loaded on canvas. The image starts moving with mouse when I try to draw any shape (Circle, Rectangle, Line etc). How can I prevent image to move when drawing?

on `mouse:down`:
```
this.shape = new fabric.Circle({
  left: pointer.x,
  top: pointer.y,
  originX: 'left',
  originY: 'top',
  radius: pointer.x - this.drawShape.origX,
  angle: 0,
  fill: this.drawShape.isInFilled ? this.drawShape.color : '',
  stroke: this.drawShape.color,
  strokeWidth: 3,
});
```
On `mouse:move`:
```
 let radius = Math.abs(this.drawShape.origY - pointer.y) / 2;
  if (radius > this.shape.strokeWidth) {
    radius -= this.shape.strokeWidth / 2;
  }
  this.shape.set({ radius: radius });

  if (this.drawShape.origX > pointer.x) {
    this.shape.set({ originX: 'right' });
  } else {
    this.shape.set({ originX: 'left' });
  }
  if (this.drawShape.origY > pointer.y) {
    this.shape.set({ originY: 'bottom' });
  } else {
    this.shape.set({ originY: 'top' });
  }
```

When i start drawing the previously added Image starts moving with it. How can i prevent it?

**Comment by ShaMan123** [comment, marked as answer]
evented: false
or use lockMovementX or other flags

**Reply by ShaMan123** [reply]
or return true from image.onSelect, better off

**Reply by Saqib92** [reply]
I did `image.set({selectable: false});` when starting drawing circle then then `selectable: true` after completing circle.

**Reply by Saqib92** [reply]
should I go with `evented : false` ? or my approach is good enough?

**Reply by ShaMan123** [reply]
what works, works :)
